spec/controllers/groups_controller_spec.rb

```diff
@@ -135,7 +136,67 @@
       end
     end
 
-    describe '#rename_group'
+    describe '#rename_group' do
+      it 'should rename a group' do
+        # The id param expects grouping id, not group id.
+        expect do
+          post_as instructor, :rename_group, params: {
+            course_id: course.id,
+            assignment_id: assignment.id,
+            id: grouping.id,
+            new_groupname: 'renamed'
+          }
+        end.to change { group.reload.group_name }.from('group1').to('renamed')
+      end
+
+      context 'group renaming and repository association' do
+        # Regression test for https://github.com/MarkUsProject/Markus/issues/7215, where renaming a group in an
+        # assignment would incorrectly associate the group's repository with one from another assignment, resulting in
+        # lost submissions and incorrect repository mappings.
+        let(:assignment2) { create(:assignment, course: course) }
+        let(:grouping_with_submission) { create(:grouping_with_inviter_and_submission, assignment: assignment2) }
+        let(:grouping_without_submission) { create(:grouping, assignment: assignment2) }
+
+        before do
+          grouping_with_submission.save
```

Comment:
Instead of doing this, you can use `let!` instead of `let` to define these variables. This will define them eagerly rather than lazily, ensuring they're created for each test case.